apps/archives/views.py

```diff
@@ -264,18 +264,23 @@ def process_advanced_search(search_params):
     if title:
         docs_qs = docs_qs.filter(Q(title__icontains=title))
 
-    text = search_params.get('text')
+    text = search_params.get('text') # full text search
     if text:
-        # allows quotation marks but only extracts the string in the middle
-        match = re.match(r'^[\'"]?([a-zA-Z\d ]+)[\'"]?$', text)
-        if not match:
-            print(f"WARNING. Could not parse full text search string: {text}.")
-        else:
-            print('Match groups: ', match.groups())
-            raw_docs = Document.objects.raw(f'''SELECT * FROM doc_fts 
-                                                     WHERE text MATCH "{match.groups()[0]}";''')
-            doc_ids = [doc.id for doc in raw_docs]
-            docs_qs = docs_qs.filter(id__in=doc_ids)
+        words_q = Q()
+
+        # handle exact search terms wrapped in " or '
+        exact_searches = re.findall(r'[\'"]([a-zA-Z\d ]+)[\'"]', text)
```

Comment:
I think we shouldn't allow single quotes. It will mess up, e.g. with `"he's here"`

Also, is `[a-zA-Z\d ]` to avoid sql injections? Could we allow dashes ("short-lasting") or periods ("P. M. Morse")?

Reply:
Agreed re single quotes.

Not sure about the inner bit bc I didn't write it, but pretty sure we don't need to worry about injections bc the ORM handles it. (https://docs.djangoproject.com/en/2.2/topics/security/#sql-injection-protection) I'll change to ```.+```